### PR TITLE
TVB-2415 Handle tvb version in build step 1

### PR DESCRIPTION
**Pull request by bvalean**

### CODE REVIEW

File: scientific_library/tvb/basic/config/environment.py [hunk 1]
```diff
@@ -84,10 +84,11 @@ def is_distribution():
...
 
         try:
-            _proc = Popen(["svnversion", "."], stdout=PIPE, stderr=PIPE)
-            version = VersionSettings.parse_svn_version(_proc.communicate()[0])
+            _proc = Popen(["git", "branch", "--show-current"], stdout=PIPE)
+            branch = str(_proc.communicate()[0])
+            version = VersionSettings.fetch_current_revision(branch)
             if version:
-                # usage from SVN (deprecated)
+                # usage from git
```
Comment (liadomide) [inline scientific_library/tvb/basic/config/environment.py:91]: Considering the `try` block above (line 71) is this still useful ?

File: framework_tvb/tvb/config/profile_settings.py [hunk 2]
```diff
@@ -95,7 +95,7 @@ class TestSQLiteProfile(WebSettingsProfile):
 
     DEFAULT_STORAGE = os.path.expanduser(os.path.join('~', 'TVB_TEST'))
 
-    SVN_VERSION = 1
+    GIT_VERSION = 1
```
Comment (liadomide) [inline framework_tvb/tvb/config/profile_settings.py:98]: How about using a generic name, for example `REVISION_NUMBER` ?

File: scientific_library/tvb/basic/config/settings.py [hunk 3]
```diff
@@ -77,25 +79,33 @@ def __init__(self, manager, bin_folder):
...
         number = ''.join([ch for ch in version_string if ch.isdigit()])
         return int(number)
 
+    @staticmethod
+    def fetch_current_revision(branch):
+        url = f'https://api.github.com/repos/the-virtual-brain/tvb-root/commits'
+        params = {'per_page': 1, 'sha': branch}
+        resp = requests.get(url, params)
+        last_link = resp.links.get('last')
+        return int(last_link['url'].split('&page=')[1])
```
Comment (liadomide) [inline scientific_library/tvb/basic/config/settings.py:107]: Here maybe we could add that static number that I mentioned (10.000) with the switch from SVN numbers to GIT count.

How would this API work in different branches ? Can we ensure a unique and incremental number is returned at two succesive calls?

File: tvb_build/Jenkinsfile-Step1 [hunk 4]
```diff
@@ -38,6 +38,9 @@ pipeline {
                     bash install_full_tvb.sh
                     python build_step1.py
                 '''
+                withCredentials([usernamePassword(credentialsId: 'tvb-revision', passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')]) {
+                        sh('git commit scientific_library/tvb/basic/config/tvb.version -m "Update GIT revision number automatically from Jenkins" && git push https://${GIT_USERNAME}:${GIT_PASSWORD}@github.com/the-virtual-brain/tvb-root.git HEAD:tvb-version || true')
+                 }
```
Comment (liadomide) [inline tvb_build/Jenkinsfile-Step1:43]: Branch name will need to be changed into `master`, or can we take it from the env to be the current branch ?

File: tvb_build/build_step1.py [hunk 5]
```diff
@@ -161,46 +161,48 @@ def _copy_demos_collapsed(to_copy):
...
     """
     import tvb.basic.config
     from tvb.basic.config.settings import VersionSettings
     config_folder = os.path.dirname(os.path.abspath(tvb.basic.config.__file__))
+    branch = "master"
 
-    svn_variable = 'SVN_REVISION'
-    if svn_variable in os.environ:
-        real_svn_number = int(os.environ[svn_variable])
+    tvb_version_variable = 'TVB_REVISION'
```
Comment (liadomide) [inline tvb_build/build_step1.py:173]: Do we ever have this env variable set `TVB_REVISION` ?

**Review by liadomide — COMMENTED** [review]
I like the direction, still I have few questions bellow

File: tvb_build/Jenkinsfile-Step1 (hunk 4 above)
Comment (bvalean) [reply tvb_build/Jenkinsfile-Step1:45]: After the current changes we are taking the branch name automatically.

File: tvb_build/build_step1.py [hunk 6]
```diff
@@ -161,46 +163,43 @@ def _copy_demos_collapsed(to_copy):
...
-        print("Updating tvb.version content to: %s because %d != %d" % (new_text, written_svn_number, real_svn_number))
-
-    # Update SVN_REVISION in the current build, as we are creating a new commit
-    os.environ[svn_variable] = str(real_svn_number + 1)
-    _proc = Popen(["svn", "commit", "../scientific_library/tvb/basic/config/tvb.version", "-m",
-                   "Update SVN revision number automatically from Hudson", "--trust-server-cert"], stdout=PIPE)
-    print(_proc.communicate()[0])
+    tvb_version_path = "../scientific_library/tvb/basic/config/tvb.version"
+    print("Tvb version file path: {}".format(tvb_version_path))
+    paths = [tvb_version_path, os.path.join(config_folder, 'tvb.version')]
```
Comment (popaula937) [inline tvb_build/build_step1.py:193]: Why is this necessary?
